### PR TITLE
KTOR-3807 Fix native flaky tests

### PR DESCRIPTION
**Pull request by e5l**
**Subsystem**
Client/Server, related modules

**Motivation**
Describe what problem this PR solves and why it is important. Refer to a bug/ticket #.

**Solution**
Describe your solution.

### CODE REVIEW

File: ktor-client/ktor-client-cio/jvmAndNix/src/io/ktor/client/engine/cio/Endpoint.kt [hunk 1]
```diff
@@ -45,35 +44,32 @@ internal class Endpoint(
...
+        }
+
         val response = CompletableDeferred<HttpResponseData>()
         val task = RequestTask(request, response, callContext)
-        processTask(task)
-        return response.await()
+        try {
+            makePipelineRequest(task)
+            return response.await()
+        } catch (cause: Throwable) {
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-cio/jvmAndNix/src/io/ktor/client/engine/cio/Endpoint.kt:69]: @sonatype-lift ignore

File: ktor-client/ktor-client-cio/jvmAndNix/src/io/ktor/client/engine/cio/Endpoint.kt [hunk 2]
```diff
@@ -119,37 +115,19 @@ internal class Endpoint(
...
 
-            val responseData = handleTimeout(timeout) {
-                writeRequestAndReadResponse(request, output, callContext, input, originOutput)
-            }
-
-            response.complete(responseData)
+            val requestTime = GMTDate()
+            writeRequest(request, output, callContext, proxy != null)
+            return readResponse(requestTime, request, input, originOutput, callContext)
         } catch (cause: Throwable) {
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-cio/jvmAndNix/src/io/ktor/client/engine/cio/Endpoint.kt:123]: @sonatype-lift ignore

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt [hunk 3]
```diff
@@ -18,7 +18,7 @@ import kotlin.test.*
 
 private const val TEST_URL = "$TEST_SERVER/timeout"
 
-class HttpTimeoutTest : ClientLoader() {
+class HttpTimeoutTest : ClientLoader(timeoutSeconds = 1000000) {
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt:21]: probably not intended

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt [hunk 4]
```diff
@@ -426,19 +426,6 @@ class HttpTimeoutTest : ClientLoader() {
         }
     }
 
-    @Test
-    fun testConnectTimeout() = clientTests(listOf("Js", "Darwin", "CIO")) {
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt:430]: does it make sense to rewrite them to use local server?

**Review by rsinukov — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt (hunk 3 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt:21]: Thanks, fixed

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingMockedTests.kt [hunk 5]
```diff
@@ -62,71 +62,6 @@ class LoggingMockedTests {
         }
     }
 
-    @Test
-    fun testLogResponseWithException() = testWithEngine(MockEngine) {
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingMockedTests.kt:66]: accidentally, fixed one more issue. thanks

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingTest.kt [hunk 6]
```diff
@@ -249,79 +249,6 @@ class LoggingTest : ClientLoader() {
         }
     }
 
-    @Test
-    fun testLogRedirect() = clientTests(listOf("js", "Curl", "CIO", "Java")) {
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/LoggingTest.kt:253]: cause responses can be logged in any order

File: ktor-io/api/ktor-io.api [hunk 7]
```diff
@@ -68,7 +68,6 @@ public abstract class io/ktor/utils/io/ByteChannelSequentialBase : io/ktor/utils
 	public fun readUTF8LineTo (Ljava/lang/Appendable;ILkotlin/coroutines/Continuation;)Ljava/lang/Object;
 	public fun request (I)Lio/ktor/utils/io/core/internal/ChunkBuffer;
 	protected final fun setClosed (Z)V
-	public final fun setClosedCause (Ljava/lang/Throwable;)V
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-io/api/ktor-io.api:71]: It looks like an internal bridge, will check

File: ktor-io/common/src/io/ktor/utils/io/core/BufferFactory.kt [hunk 8]
```diff
@@ -68,6 +67,10 @@ public class DefaultBufferPool(
     override fun validateInstance(instance: ChunkBuffer) {
         super.validateInstance(instance)
 
+        check(instance.memory.size32 == bufferSize) {
+            "Buffer size mismatch. Expected: $bufferSize, actual: ${instance.memory.size}"
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-io/common/src/io/ktor/utils/io/core/BufferFactory.kt:71]: It doesn't matter in this case, will use size everywhere

File: ktor-utils/posix/src/io/ktor/util/CoroutineUtils.kt [hunk 9]
```diff
@@ -34,8 +34,10 @@ private class MultiWorkerDispatcher(name: String, workersCount: Int) : Closeable
     }
 
     private fun workerRunLoop() = runBlocking {
-        for (task in tasksQueue) {
-            task.run()
+        kotlin.runCatching {
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-utils/posix/src/io/ktor/util/CoroutineUtils.kt:41]: it should be in the origin task

File: ktor-io/api/ktor-io.api (hunk 7 above)
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-io/api/ktor-io.api]: Fixed

**Review by rsinukov — APPROVED** [review]

**Pull request merged**